user/db.go

```diff
@@ -73,161 +70,276 @@ func (s *Store) updateUser(ctx context.Context, user User) error {
 	if err != nil {
 		return nil, err
 	}
 	return u, nil
 }
 
-// getUsersUA will get a group of users by username ASC
-func (s *Store) getUsersUA(ctx context.Context) ([]User, error) {
+// getUsersSearchOptionsAsc will get users search with sorting asc
+func (s *Store) getUsersSearchOptionsAsc(ctx context.Context, search, sortBy string) ([]User, error) {
 	var u []User
 	err := s.db.SelectContext(ctx, &u,
-		`SELECT user_id, username, nickname, first_name, last_name, email, last_login, avatar, use_gravatar
+		`SELECT *
 		FROM people.users
-		ORDER BY username ASC;`)
+		WHERE (CAST(user_id AS TEXT) LIKE '%' || $1 || '%')
+		   OR (username LIKE '%' || $1 || '%')
+		   OR (nickname LIKE '%' || $1 || '%')
+		   OR (first_name LIKE '%' || $1 || '%')
+		   or (last_name LIKE '%' || $1 || '%')
+		   OR (email LIKE '%' || $1 || '%')
+		   OR (first_name || ' ' || last_name LIKE '%' || $1 || '%')
+		ORDER BY
+		    CASE WHEN $2 = 'userId' THEN user_id END ASC,
+		    CASE WHEN $2 = 'name' THEN first_name END ASC,
+			CASE WHEN $2 = 'name' THEN last_name END ASC,
+		    CASE WHEN $2 = 'username' THEN username END ASC,
+		    CASE WHEN $2 = 'email' THEN email END ASC,
+		    CASE WHEN $2 = 'lastLogin' THEN last_login END ASC NULLS FIRST;`, search, sortBy)
 	if err != nil {
 		return nil, err
 	}
 	return u, nil
 }
 
-// getUsersUD will get a group of users by username DESC
-func (s *Store) getUsersUD(ctx context.Context) ([]User, error) {
+// getUsersSearchOptionsAscSizePage will get users search with sorting asc with size and page
+func (s *Store) getUsersSearchOptionsAscSizePage(ctx context.Context, search, sortBy string, size, page int) ([]User, error) {
 	var u []User
 	err := s.db.SelectContext(ctx, &u,
-		`SELECT user_id, username, nickname, first_name, last_name, email, last_login, avatar, use_gravatar
+		`SELECT *
 		FROM people.users
-		ORDER BY username DESC;`)
+		WHERE (CAST(user_id AS TEXT) LIKE '%' || $1 || '%')
+		   OR (username LIKE '%' || $1 || '%')
+		   OR (nickname LIKE '%' || $1 || '%')
+		   OR (first_name LIKE '%' || $1 || '%')
+		   or (last_name LIKE '%' || $1 || '%')
+		   OR (email LIKE '%' || $1 || '%')
+		   OR (first_name || ' ' || last_name LIKE '%' || $1 || '%')
+		ORDER BY
+		    CASE WHEN $2 = 'userId' THEN user_id END ASC,
+		    CASE WHEN $2 = 'name' THEN first_name END ASC,
+			CASE WHEN $2 = 'name' THEN last_name END ASC,
+		    CASE WHEN $2 = 'username' THEN username END ASC,
+		    CASE WHEN $2 = 'email' THEN email END ASC,
+		    CASE WHEN $2 = 'lastLogin' THEN last_login END ASC NULLS FIRST
+		LIMIT $3
+		OFFSET $4;`, search, sortBy, size, size*(page-1))
 	if err != nil {
 		return nil, err
 	}
 	return u, nil
 }
 
-// getUsersEA will get a group of users by email ASC
-func (s *Store) getUsersEA(ctx context.Context) ([]User, error) {
+// getUsersOptionsDesc will get users sorting desc
+func (s *Store) getUsersOptionsDesc(ctx context.Context, sortBy string) ([]User, error) {
```

Comment:
Could the `getUsersXxxAsc` and `getUsersXxxDesc` functions be combined, or does the SQL get gnarly? (I'm worried about the possibility that if we add another sort option then you need to update all of these.)

Reply:
That is exactly how it got so again a very much future fix does this